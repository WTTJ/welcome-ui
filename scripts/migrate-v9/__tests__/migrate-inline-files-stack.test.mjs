import { readFileSync } from 'fs'
import { dirname, resolve } from 'path'
import { fileURLToPath } from 'url'

import { describe, expect, it } from 'vitest'

import { copyDirSync, deleteDirRecursive } from '../helpers/file-utils.mjs'
import { migrateAll } from '../index.mjs'

const __dirname = dirname(fileURLToPath(import.meta.url))

describe('Inline Migration - Stack Components', () => {
  it('migrates Stack components with layout props to div with classNames', async () => {
    const fixturesDir = resolve(__dirname, './__fixtures__')
    const tempDir = resolve(__dirname, '../temp/inline-stack-test')

    // Clean up and copy fixtures
    deleteDirRecursive(tempDir)
    copyDirSync(fixturesDir, tempDir)

    // Run migration
    await migrateAll(tempDir, { copyDir: false, interactive: false })

    // Read results
    const migratedComponent = readFileSync(resolve(tempDir, 'StackComponent.tsx'), 'utf8')

    // Verify with snapshots
    expect(migratedComponent).toMatchSnapshot('stack-component.tsx')

    // Cleanup
    deleteDirRecursive(tempDir)
  })
})
