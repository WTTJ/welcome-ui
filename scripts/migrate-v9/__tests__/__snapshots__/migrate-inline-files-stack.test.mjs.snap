// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Inline Migration - Stack Components > migrates Stack components with layout props to div with classNames > stack-component.tsx 1`] = `
"import { Stack } from 'welcome-ui/Stack'

export const StackComponent = () => {
  return (
    <div>
      <div className="flex flex-row gap-xxs items-center">
        <div>Stack item 1</div>
        <div>Stack item 2</div>
      </div>
      <div className="flex flex-1 flex-col gap-xxl">
        <div>Flexible stack item</div>
      </div>
    </div>
  )
}
"
`;
